src/util.h

```diff
@@ -26,10 +26,13 @@
 #include <string>
 #include <vector>
 
-#ifdef _MSC_VER
-#define NORETURN __declspec(noreturn)
-#else
-#define NORETURN __attribute__((noreturn))
+#ifdef __has_cpp_attribute
+#if __has_cpp_attribute(noreturn)
+#define NORETURN [[noreturn]]
+#endif
+#endif
+#ifndef NORETURN
+#define NORETURN  // old compilers
```

Comment:
Thank you, can you make this a little bit more readable with indentation, and less nesting, e.g.:

```c++
#if !defined(__has_cpp_attribute)
#  define __has_cpp_attribute(x)  0
#endif

#if __has_cpp_attribute(noreturn)
#  define NORETURN [[noreturn]]
#else
#  define NORETURN  // nothing for old compilers
#endif
```